# HG changeset patch
# User Dominique Leuenberger <dimstar@opensuse.org>
# Date 1468914700 -7200
#      Tue Jul 19 09:51:40 2016 +0200
# Node ID 434e1ff5323789560c210b0b417844b8ab232ab0
# Parent  24fc93301077a2317d1d6deda3b7d895e6754e26
build: Fix linking with new xcb versions

As we make use of specific xcb and xcb_shm functions, it is our sole responsibility
to ensure we link against the right libs. Expecting xcb-shape to bring in all the
dependencies for us is wrong (and no longer works in all cases)

diff -r 24fc93301077 -r 434e1ff53237 m4/video_out.m4
--- a/m4/video_out.m4	Tue Jun 07 19:33:35 2016 +0200
+++ b/m4/video_out.m4	Tue Jul 19 09:51:40 2016 +0200
@@ -305,7 +305,7 @@
     dnl xcb
     XINE_ARG_WITH([xcb], [Enable support for XCB video out plugins])
     if test x"$with_xcb" != x"no"; then
-        PKG_CHECK_MODULES([XCB], [xcb-shape >= 1.0], [have_xcb=yes], [have_xcb=no])
+        PKG_CHECK_MODULES([XCB], [xcb xcb-shape >= 1.0], [have_xcb=yes], [have_xcb=no])
         if test x"$hard_enable_xcb" = x"yes" && test x"$have_xcb" != x"yes"; then
             AC_MSG_ERROR([XCB support requested, but XCB not found])
         elif test x"$have_xcb" = x"yes"; then
diff -r 24fc93301077 -r 434e1ff53237 src/video_out/Makefile.am
--- a/src/video_out/Makefile.am	Tue Jun 07 19:33:35 2016 +0200
+++ b/src/video_out/Makefile.am	Tue Jul 19 09:51:40 2016 +0200
@@ -147,7 +147,7 @@
 xineplug_vo_out_xcbshm_la_CFLAGS = $(AM_CFLAGS) $(XCB_CFLAGS) $(XCBSHM_CFLAGS) $(AVUTIL_CFLAGS)
 
 xineplug_vo_out_xcbxv_la_SOURCES = video_out_xcbxv.c $(XCBOSD)
-xineplug_vo_out_xcbxv_la_LIBADD = $(XINE_LIB) $(PTHREAD_LIBS) $(LTLIBINTL) $(XCBXV_LIBS) $(XCB_LIBS)
+xineplug_vo_out_xcbxv_la_LIBADD = $(XINE_LIB) $(PTHREAD_LIBS) $(LTLIBINTL) $(XCBXV_LIBS) $(XCB_LIBS) $(XCBSHM_LIBS)
 xineplug_vo_out_xcbxv_la_CFLAGS = $(AM_CFLAGS) $(XCB_CFLAGS) $(XCBXV_CFLAGS)
 
 xineplug_vo_out_xshm_la_SOURCES = video_out_xshm.c $(X11OSD)
